Decred Wallet/Features/Navigation Menu/TabNavigationMenu.swift

```diff
@@ -0,0 +1,126 @@
+        let itemIconView = UIImageView(frame: CGRect.zero)
+        
+        itemTitleLabel.font = UIFont(name: "Source Sans Pro", size: 13)
+        itemTitleLabel.text = item.displayTitle
+        itemTitleLabel.textColor = UIColor.appColors.darkGray
+        itemTitleLabel.textAlignment = .center
+        itemTitleLabel.translatesAutoresizingMaskIntoConstraints = false
+        itemTitleLabel.clipsToBounds = true
+        
+        itemIconView.image = item.icon!.withRenderingMode(.automatic)
+        itemIconView.translatesAutoresizingMaskIntoConstraints = false
+        itemIconView.clipsToBounds = true
+        
+        tabBarItem.layer.backgroundColor = UIColor.white.cgColor
+        tabBarItem.addSubview(itemIconView)
+        tabBarItem.addSubview(itemTitleLabel)
+        tabBarItem.translatesAutoresizingMaskIntoConstraints = false
+        tabBarItem.clipsToBounds = true
+        
+        let constraints = [
+            itemIconView.heightAnchor.constraint(equalToConstant: 25), // Fixed height for our tab item
+            itemIconView.widthAnchor.constraint(equalToConstant: 25), // Fixed width for our tab item
+            itemIconView.centerXAnchor.constraint(equalTo: tabBarItem.centerXAnchor),
+            itemIconView.topAnchor.constraint(equalTo: tabBarItem.topAnchor, constant: 8), // Position menu item icon 8pts from the top of it's parent view
+            itemIconView.leadingAnchor.constraint(equalTo: tabBarItem.leadingAnchor, constant: 35),
+            itemTitleLabel.heightAnchor.constraint(equalToConstant: 13), // Fixed height for title label
+            itemTitleLabel.widthAnchor.constraint(equalTo: tabBarItem.widthAnchor), // Position label full width across tab bar item
+            itemTitleLabel.topAnchor.constraint(equalTo: itemIconView.bottomAnchor, constant: 4), // Position title label 4pts below item icon
+        ]
+        NSLayoutConstraint.activate(constraints)
+        tabBarItem.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.handleTap))) // Each item should be able to trigger and action on tap
+        return tabBarItem
+    }
+    
+    @objc func handleTap(_ gesture: UIGestureRecognizer) {
+        self.switchTab(from: self.activeItem, to: gesture.view!.tag)
+    }
+    
+    public func switchTab(from: Int, to: Int) {
+        self.deactivateTab(viewId: from)
+        self.activateTab(viewId: to)
+    }
+    
+    func deactivateTab(viewId: Int) {
+        let tab = self.subviews[viewId]
+        let layersToRemove = tab.layer.sublayers!.filter({ $0.name == "active border" })
+        
+        DispatchQueue.main.async {
+            UIView.animate(withDuration: 0.4, delay: 0.0, options: [.curveEaseIn, .allowUserInteraction], animations: {
+                layersToRemove.forEach({ $0.removeFromSuperlayer() })
+                tab.setNeedsLayout()
+                tab.layoutIfNeeded()
+            })
+        }
+    }
+    
+    func activateTab(viewId: Int) {
```

Comment:
This function can make use of 1 or more empty line spaces to make it more readable.

Reply:
resolved in 72e13de